sources/packages/backend/apps/workers/src/controllers/program-info-request/program-info-request.controller.ts

```diff
@@ -0,0 +1,61 @@
+import { Controller } from "@nestjs/common";
+import { ZeebeWorker } from "../../zeebe";
+import { ZeebeJob, MustReturnJobActionAcknowledgement } from "zeebe-node";
+import { ApplicationService } from "../../services";
+import {
+  ProgramInfoRequestJobInDTO,
+  ProgramInfoRequestJobHeaderDTO,
+  ProgramInfoRequestJobOutDTO,
+} from "..";
+import { APPLICATION_ID } from "../workflow-variables";
+import { APPLICATION_NOT_FOUND } from "../error-code-constants";
+
+@Controller()
+export class ProgramInfoRequestController {
+  constructor(private readonly applicationService: ApplicationService) {}
+
+  /**
+   * Defines the Program Information Request (PIR) status for the student
+   * application returning the its most updated status.
+   * @returns most updated status of the PIR.
+   */
+  @ZeebeWorker("program-info-request", { fetchVariable: [APPLICATION_ID] })
+  async updateApplicationStatus(
+    job: Readonly<
+      ZeebeJob<
+        ProgramInfoRequestJobInDTO,
+        ProgramInfoRequestJobHeaderDTO,
+        ProgramInfoRequestJobOutDTO
+      >
+    >,
+  ): Promise<MustReturnJobActionAcknowledgement> {
+    const application = await this.applicationService.getApplicationById(
+      job.variables.applicationId,
+      { loadDynamicData: false },
+    );
+    if (!application) {
+      return job.error(
+        APPLICATION_NOT_FOUND,
+        "Application not found while verifying the PIR.",
+      );
+    }
+    if (application.pirStatus) {
+      // PIR status was already set, just return it.
+      return job.complete({
+        programInfoStatus: application.pirStatus,
+      });
+    }
+    const updateResult = await this.applicationService.updateProgramInfoStatus(
+      job.variables.applicationId,
+      job.customHeaders.programInfoStatus,
+    );
+    if (updateResult.affected) {
+      return job.complete({
+        programInfoStatus: job.customHeaders.programInfoStatus,
+      });
+    }
+    return job.fail(
```

Comment:
Error description can be asjusted here. I know error here is only for technical troubleshooting/analysis mostly but it can be updated as either application not found or PIR could not be updated.

Reply:
Adjusted.